integration-templates/zoho-crm/zoho-crm-accounts.ts

```diff
@@ -0,0 +1,124 @@
+import type { ZohoCRMAccount, NangoSync } from './models';
+
+export default async function fetchData(nango: NangoSync) {
+    try {
+        const responses = await getAllAccounts(nango, '/crm/v2/Accounts');
+        if (responses.length > 0) {
+            const mappedAccounts: ZohoCRMAccount[] = responses.map(mapAccounts);
+            // Save Accounts
+            await nango.batchSave(mappedAccounts, 'ZohoCRMAccount');
+        } else {
+            await nango.log('No Accounts found.');
+        }
+    } catch (error: any) {
+        await nango.log(`Error in fetchData: ${error.message}`);
+    }
+}
+
+async function getAllAccounts(nango: NangoSync, endpoint: string): Promise<any[]> {
+    const accounts: any[] = [];
+    let page = 1;
+    const perPage = 100;
+    let response: { data: { data?: any[]; info?: { more_records?: boolean }; nextPage?: string } };
+
+    try {
+        do {
+            const { params, headers } = generateApiRequestParams(nango, page, perPage);
+
+            response = await nango.get({
+                endpoint: endpoint,
+                params: params,
+                headers: headers
+            });
+
+            if (!response.data.data) {
+                throw new Error(`Received a ZohoCRMAccounts API error (for ${endpoint}): ${JSON.stringify(response.data, null, 2)}`);
+            }
+
+            accounts.push(...response.data.data);
+            page++; // Increment the page for the next request
+        } while (response.data.info?.more_records);
+
+        return accounts;
+    } catch (error: any) {
+        // Handle unexpected errors, log them, and possibly retry or take appropriate action
+        await nango.log(`Error in getAllAccounts: ${error.message}`);
```

Comment:
Best practice is to throw an exception in case of an error, so the sync fails transparently and the user knows they need to investigate.
```
throw new Error(`Error message here`);
```